geospaas_harvesting/harvesters.py

```diff
@@ -83,24 +90,58 @@ def harvest(self):
                 break
 
 
-class PODAACHarvester(Harvester):
-    """Harvester class for PODAAC data (NASA)"""
+class WebDirectoryHarvester(Harvester):
+    """
+    class for harvesting online data sources that rely on webpages (and most of the time on opendap)
+    """
+
     def _create_crawlers(self):
-        return [
-            crawlers.OpenDAPCrawler(url, time_range=(self.get_time_range()))
-            for url in self.config['urls']
-        ]
+        try:
+            assert self.crawler
+            return [
+                self.crawler(url, time_range=(self.get_time_range()))
+                for url in self.config['urls']
+            ]
+        except (KeyError, TypeError, AttributeError, AssertionError) as error:
+            if isinstance(error, AssertionError):
+                raise HarvesterConfigurationError(
+                    "The class of crawler has not been specified properly") from error
+            else:
+                raise HarvesterConfigurationError(
+                    "crawler must be created properly with correct configuration file") from error
```

Comment:
There should be only one operator inside `try`. 

Replace either with 
```suggestion
        try:
            assert self.crawler
        except AssertionError as error:
            something here
        try:
            return [
                self.crawler(url, time_range=(self.get_time_range()))
                for url in self.config['urls']
            ]
        except (KeyError, TypeError, AttributeError) as error:
            something else here
```

Or better remove `assert self.crawler`. `assert` is good only for debugging. If you want to check that `crawler` part of `self` use `hasattr`.

And never use `if` to select exception. Use multiple `except` (https://stackoverflow.com/questions/6095717/python-one-try-multiple-except)

Please fix here and below.

Reply:
Done in the next commit.

Reply:
To test the validity of the `crawler` and `ingester` attributes, you can just use `isinstance(self.crawler, crawlers.Crawler)`